docs/reference/query-builders.md

```diff
@@ -0,0 +1,382 @@
+
+### `search()`
+
+```
+query.search('tree')
+```
+
+The `search` builder limits results to those matching the string passed as an argument. Search is implemented using MongoDB's `$text` operator and a full text index.
+
+If this query builder is set, the `sort` query builder will default to sorting by search quality. This is important because the worst of the full-text search matches will be of very poor quality.
+
+### `type()`
+
+```
+query.type('product')
+```
+
+The `type` builder can be used to limit a query to only one particular doc type. Pass the doc type name (string) as an argument.
+
+Generally you don't want to call this method directly. It will be better to include the doc type name as the `type` criteria in the original `find()` method arguments.
+
+### `withPublished()`
+
+```
+query.withPublished(true)
+```
+
+If the `withPublished` builder is set to `true`, then each document in the results will include a `_publishedDoc` property. That property will be set to the published version of that document. This is really only useful when querying for draft documents.
+
+## Page document query builders
+
+The following query builders only relate to page documents.
+
+### `ancestors()`
+
+```
+query.ancestors(true)
+
+query.ancestors({ children: true })
+```
+
+Setting the `ancestors` builder to `true` retrieves the array of "ancestors" for each returned page and assigns them to the `_ancestors` property. Page ancestors are those that precede a given page in the page tree hierarchy. The home page is `_ancestors[0]`. A returned page is not included in its own `_ancestors` array.
+
+If the builder argument is an object, do all of the above, and also call the query builders present in the object *on the query that fetches the ancestors*. For example, you can pass `{ children: true }` to fetch the children of each ancestor as the `_children` property of each ancestor, or pass `{ children: { depth: 2 } }` to get two layers of child pages.
+
+`ancestors` also has its own `depth` option, but it might not do what you think. If the `depth` option is present as a top-level property of the `ancestors` builder argument, then only that number of ancestors are retrieved, counting backwards from the immediate parent of each page. So `{ depth: 2 }` retrieves only the two closest ancestors.
+
+### `children()`
+
+```
+query.children(true)
+
+query.children({ depth: 1 })
+```
+
+The `children` builder is used to include the "children" of returned pages in an array on a `_children` property. If `children(true)` is called, it will return all children of a given page. If the argument is an object, it may have a `depth` property to fetch a specific number of child layers. Any
+other properties are passed on *as builders for the query that fetches the children*.
+
+### `isPage()`
+
+```
+query.isPage(true)
+```
+
+Passing `true` to the `isPage` builder ensures that results will only include documents that are pages. It defaults to `true` when used.
```

Comment:
It defaults to `true`. There's no "when used". It's just that the builder only exists in a query that came from `apos.page.find`.